src/components/TracePage/TraceTimelineViewer/TimelineHeaderRow.js

```diff
@@ -0,0 +1,190 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as React from 'react';
+import cx from 'classnames';
+
+import Ticks from './Ticks';
+import TimelineRow from './TimelineRow';
+
+import './TimelineHeaderRow.css';
+
+type TimelineHeaderRowProps = {
+  endTime: number,
+  nameColumnWidth: number,
+  numTicks: number,
+  onColummWidthChange: number => void,
+  startTime: number,
+};
+
+type TimelineColumnResizerProps = {
+  min: number,
+  max: number,
+  onChange: number => void,
+  position: number,
+};
+
+type TimelineColumnResizerState = {
+  currentDragPosition: ?number,
+  rootX: ?number,
+};
+
+const LEFT_MOUSE_BUTTON = 0;
+
+class TimelineColumnResizer extends React.PureComponent<
+  TimelineColumnResizerProps,
+  TimelineColumnResizerState
+> {
+  props: TimelineColumnResizerProps;
+  state: TimelineColumnResizerState;
+  _rootElm: ?Element;
+  _isDragging: boolean;
+
+  constructor(props) {
+    super(props);
+    this._rootElm = undefined;
+    this._isDragging = false;
+    this.state = {
+      currentDragPosition: null,
+      rootX: null,
+    };
+    this._setRootElm = this._setRootElm.bind(this);
+    this._onDraggerMouseDown = this._onDraggerMouseDown.bind(this);
+    this._onWindowMouseMove = this._onWindowMouseMove.bind(this);
+    this._onWindowMouseUp = this._onWindowMouseUp.bind(this);
+  }
+
+  componentWillUnmount() {
+    if (this._isDragging) {
+      window.removeEventListener('mousemove', this._onWindowMouseMove);
+      window.removeEventListener('mouseup', this._onWindowMouseUp);
+      this._isDragging = false;
+    }
+  }
+
+  _setRootElm = function _setRootElm(elm) {
+    this._rootElm = elm;
+  };
+
+  _onDraggerMouseDown = function _onDraggerMouseDown({ button, clientX }) {
+    if (this._isDragging || button !== LEFT_MOUSE_BUTTON || !this._rootElm) {
+      return;
+    }
+    const rootX = this._rootElm.getBoundingClientRect().left;
+    this.setState({
+      rootX,
+      currentDragPosition: clientX - rootX,
+    });
+    window.addEventListener('mousemove', this._onWindowMouseMove);
+    window.addEventListener('mouseup', this._onWindowMouseUp);
+    this._isDragging = true;
+    if (document && document.body && document.body.style) {
+      (document.body.style: any).userSelect = 'none';
+    }
+  };
+
+  _onWindowMouseMove = function _onWindowMouseMove({ clientX }) {
+    const { rootX } = this.state;
+    this.setState({ rootX, currentDragPosition: clientX - (rootX || 0) });
+  };
+
+  _onWindowMouseUp = function _onWindowMouseUp({ clientX }) {
+    window.removeEventListener('mousemove', this._onWindowMouseMove);
+    window.removeEventListener('mouseup', this._onWindowMouseUp);
+    if (document && document.body && document.body.style) {
```

Comment:
You can simplify with lodash `_.get(document, 'style.body')`

Reply:
Thanks. I'd like to use [idx](https://github.com/facebookincubator/idx), but it's not worth ejecting...